### PR TITLE
Login form UI redesign

### LINKED ISSUE #36234: New login flow

### PR DESCRIPTION
**Pull request by bl-nero**
Please note: I used a new library in the test (`react-select-event`), and installing it apparently made Lint explode, pulled in a ton of spurious deps, and triggered a vulnerability check. I'll deal with it once I'm back from vacation next week, but the rest of the code is ready for review.

Closes #36234

Demo: https://www.loom.com/share/93edb868557246afba63d76c2469662c?sid=2e596480-4845-48c8-8869-98d673dd285a

Tested:
- Passwordless (success and failure; primary and secondary)
- Local form with passkey (success and failure; secondary)
- Local form with WebAuthn MFA (secondary)
- Local form with OTP (success and failure; primary, secondary)
- SSO (GitHub, success and failure)

Changelog: Redesigned the login UI.

### CODE REVIEW

**Comment by bl-nero** [comment]
Ok, looks like all that yarn.lock mess was just my local Yarn going crazy and installing stuff all over the place; it's OK now.

File: web/packages/shared/utils/createMfaOptions.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import { Auth2faType, PreferredMfaType } from 'shared/services/types';
 
 export default function createMfaOptions(opts: Options) {
+  console.log('CREATING MFA OPTIONS', opts);
```
Comment (rudream) [inline web/packages/shared/utils/createMfaOptions.ts:22]:
```suggestion
```

Looks like you forgot to delete

File: web/packages/shared/utils/createMfaOptions.ts [hunk 2]
```diff
@@ -40,6 +41,7 @@ export default function createMfaOptions(opts: Options) {
     mfaOptions.push({ value: 'optional', label: 'None' });
   }
 
+  console.log(mfaOptions);
```
Comment (rudream) [inline web/packages/shared/utils/createMfaOptions.ts:44]:
```suggestion
```

File: web/packages/teleport/src/components/FormLogin/FormLogin.tsx [hunk 3]
```diff
@@ -346,127 +342,76 @@ const LocalForm = ({
...
-          {...otherProps}
-          hasTransitionEnded={hasTransitionEnded}
-          autoFocus={true}
-        />
+      return primary ? (
+        <LocalForm {...otherProps} autoFocus={true} />
+      ) : (
+        <Box py={2}>
+          <ButtonSecondary size="large" block onClick={next}>
+            Sign in with Username and Password
```
Comment (rudream) [inline web/packages/teleport/src/components/FormLogin/FormLogin.tsx:399]:
```suggestion
            Sign in with username and password
```

File: web/packages/teleport/src/components/FormLogin/FormLogin.tsx [hunk 4]
```diff
@@ -48,67 +49,65 @@ import { StepSlider, StepComponentProps } from 'design/StepSlider';
...
-        {attempt.isFailed && (
-          <Alerts.Danger m={5} mb={0}>
-            {attempt.message}
-          </Alerts.Danger>
-        )}
-        <SsoList {...props} autoFocus={true} hasTransitionEnded={true} />
-      </Card>
-    );
-  }
+  const actuallyPrimaryType = isLocalAuthEnabled ? primaryAuthType : 'sso';
```
Comment (rudream) [inline web/packages/teleport/src/components/FormLogin/FormLogin.tsx:71]:
```suggestion
  const actualPrimaryType = isLocalAuthEnabled ? primaryAuthType : 'sso';
```

File: web/packages/teleport/src/components/PasskeyIcons/index.ts
Comment (rudream) [inline web/packages/teleport/src/components/PasskeyIcons/index.ts:1]: nit: If this is just a grouping of icons only meant to be used by `AddAuthDeviceWizard`, then our current pattern would be to just keep it in that same directory and reserve `teleport/components` for components that are used by multiple components.

File: web/packages/teleport/src/components/FormLogin/FormLogin.tsx [hunk 5]
```diff
@@ -48,67 +49,65 @@ import { StepSlider, StepComponentProps } from 'design/StepSlider';
...
+        <StepSlider<typeof loginViews>
+          flows={loginViews}
+          currFlow={'default'}
+          otherAuthTypes={otherAuthTypes}
+          {...props}
+          primaryAuthType={actuallyPrimaryType}
+        />
+      ) : (
+        <Text mx={4} typography="paragraph2">
+          The ability to login has not been enabled. Please contact your system
```
Comment (rudream) [inline web/packages/teleport/src/components/FormLogin/FormLogin.tsx:107]: Out of curiosity, is this actually possible?

Comment (bl-nero) [reply web/packages/teleport/src/components/FormLogin/FormLogin.tsx:107]: That's a very good question. I could have sworn I've seen this message once, but now I don't remember if it was actually in the wild, or just in the storybook.

Comment (bl-nero) [reply web/packages/teleport/src/components/FormLogin/FormLogin.tsx:107]: OK, I found the case. It is possible to configure (or, should I say, misconfigure) the system the following way:

```yaml
auth_service:
  authentication:
    type: local
    local_auth: false
```

File: web/packages/teleport/src/components/PasskeyIcons/index.ts
Comment (bl-nero) [reply web/packages/teleport/src/components/PasskeyIcons/index.ts:1]: It used to be only used by the wizard, but now, I'm also reusing it in the login form, and that's why I moved it to `teleport/components`.

**Review by ryanclark — APPROVED** [review]

**Review by rudream — APPROVED** [review]

**Comment by bl-nero** [comment]
Tested:
- Passwordless (success and failure; primary, secondary, enterprise)
- Local form with passkey (success and failure; primary, secondary, enterprise)
- Local form with WebAuthn MFA (success and failure; primary, secondary, enterprise)
- Local form with OTP (success and failure; secondary)
- Local form without MFA (success and failure; secondary)
- SSO (GitHub, success and failure, enterprise)

File: yarn.lock [hunk 6]
```diff
@@ -3487,6 +3487,20 @@
...
+    "@babel/runtime" "^7.12.5"
+    "@types/aria-query" "^5.0.1"
+    aria-query "5.1.3"
+    chalk "^4.1.0"
+    dom-accessibility-api "^0.5.9"
+    lz-string "^1.5.0"
+    pretty-format "^27.0.2"
+
 "@testing-library/dom@^9.0.0":
   version "9.3.3"
```
Comment (ravicious) [inline yarn.lock:3505]: This results in the wrong `@testing-library/dom` being resolved by yarn which then causes a ton of `act` warnings in tests, see #33356 and [a slack thread](https://gravitational.slack.com/archives/C02DQ1C2BMW/p1703063417879609?thread_ts=1703028787.728229&cid=C02DQ1C2BMW) from the previous time this has happened.

I'll post a PR which fixes that soon.

Comment (bl-nero) [reply yarn.lock:3505]: @ravicious Thanks, I wasn't aware of this issue.

**Pull request merged**